crates/nu-cli/src/completion/matchers/fuzzy.rs

```diff
@@ -0,0 +1,25 @@
+use crate::completion::matchers;
+
+pub struct Matcher;
+
+impl matchers::Matcher for Matcher  {
+    fn matches (
+        &self,
+        partial: &str,
+        from: &str
+    ) -> bool {
+        //FIXME: get complete char list, find out other fuzzy matcher algos
+        let ignore_chars = "- _,.,!~ /\\";
+        let from_replaced = drop_chars_in_str(from, ignore_chars);
+        let partial_replaced = drop_chars_in_str(partial, ignore_chars);
+        
+        let unicode_matcher: Box<dyn matchers::Matcher> = Box::new(matchers::unicode_case_insensitive::Matcher);
+        unicode_matcher.matches(partial_replaced.as_str(), from_replaced.as_str())
```

Comment:
tl;dr I think we can probably drop this Matcher in this PR

---

I think it would be good to perhaps just drop this matcher for now. I like PRs to be as small and cohesive as possible. Applying that approach here, we have a fairly incomplete matcher that doesn't feel like it fits the concept of "fuzzy". What we can do is come back later and add a fully-featured fuzzy implementation.

Note, my idea of "fuzzy" would be that `partial` is a subsequence of `from`. For example, `abc` would match `a123b456c`. I think this is how `fzf` works, for example.

Reply:
OK, I will remove that and look into making fuzzy better.